web/src/main/java/com/microsoft/applicationinsights/web/internal/httputils/HttpServerHandler.java

```diff
@@ -112,6 +114,11 @@ public void handleEnd(ServletRequest request, ServletResponse response,
         int resultCode = extractor.getStatusCode(response);
         requestTelemetry.setSuccess(resultCode < 400);
         requestTelemetry.setResponseCode(Integer.toString(resultCode));
```

Comment:
I think these three lines above can go in a conditional `... == null` and then the `if` below can be `else`